### PR TITLE
fix: configure ocim course export [BB-5146]

### PR DESCRIPTION
**Pull request by kaustavb12**
## Description

Changes to the export scripts and their requirements to enable script to run in OCIM

## Supporting information

[BB-5146](https://tasks.opencraft.com/browse/BB-5146)

### CODE REVIEW

File: exporter/tasks.py [hunk 1]
```diff
@@ -266,59 +274,44 @@ def run(cls, filename, dry_run, **kwargs):
...
+            try:
+                log.info(f"Checking if marker file {s3_marker_filename} exsists")
+                s3_client.head_object(Bucket=kwargs['pipeline_bucket'], Key=s3_marker_filename)
+            except botocore.exceptions.ClientError as e:
+                error_code = e.response['Error']['Code']
+                if error_code == "404":
+                    error_message = f"Unable to find success marker for export {s3_marker_filename}"
+                else:
+                    error_message = f'Got {error_code} error while checking marker file'
+                raise Exception(error_message)
```
Comment (0x29a) [inline exporter/tasks.py:296]: Have you replaced `FatalTaskError` with `Exception` intentionally?

File: exporter/tasks.py [hunk 2]
```diff
@@ -266,59 +274,44 @@ def run(cls, filename, dry_run, **kwargs):
...
-            # as that will happen when a particular database table is empty.
+                log.info(f"Checking if source file {s3_source_filename} exsists")
+                s3_client.head_object(Bucket=kwargs['pipeline_bucket'], Key=s3_source_filename)
+            except botocore.exceptions.ClientError as e:
+                error_code = e.response['Error']['Code']
+                if error_code == "404":
+                    error_message = f"Unable to find success source for export {s3_source_filename}"
+                else:
+                    error_message = f'Got {error_code} error while checking source file'
+                raise Exception(error_message)
```
Comment (0x29a) [inline exporter/tasks.py:307]: Same about `FatalTaskError`.

File: exporter/tasks.py [hunk 3]
```diff
@@ -266,59 +274,44 @@ def run(cls, filename, dry_run, **kwargs):
...
-                    local_kwargs = dict(**kwargs)
-                    local_kwargs['max_tries'] = MAX_TRIES_FOR_COPY_FILE_FROM_S3
-                    execute_shell(cmd, **local_kwargs)
-                except subprocess.CalledProcessError:
-                    log.error('Unable to copy %s to %s', s3_source_filename, filename)
-                    raise
+                log.info(f"Downloading source file {s3_source_filename} into {filename}")
+                s3_client.download_file(kwargs['pipeline_bucket'], s3_source_filename, filename)
+            except:
+                raise Exception(f"Unable to copy {s3_source_filename} to {filename}")
```
Comment (0x29a) [inline exporter/tasks.py:313]: Same about `FatalTaskError`.

File: exporter/tasks.py [hunk 4]
```diff
@@ -168,35 +176,38 @@ class MongoTask(Task):
...
-      --collection {mongo_collection}
-      --query '{query}'
-      --slaveOk
-      --out {filename}
-      >&2
-    """
+
+    @classmethod
+    def constructMongoURI(*args, **kwargs):
+
```
Comment (0x29a) [inline exporter/tasks.py]: Nit: redundant newline.
```suggestion
```

File: exporter/tasks.py [hunk 5]
```diff
@@ -266,59 +274,44 @@ def run(cls, filename, dry_run, **kwargs):
...
-            )
+            s3_client = boto3.client('s3')
 
-            source_command = head_command.format(
-                bucket=kwargs['pipeline_bucket'],
-                key=s3_source_filename
-            )
+            try:
+                log.info(f"Checking if marker file {s3_marker_filename} exsists")
+                s3_client.head_object(Bucket=kwargs['pipeline_bucket'], Key=s3_marker_filename)
```
Comment (0x29a) [inline exporter/tasks.py:289]: Nice, thanks for refactoring this. :+1:

File: exporter/tasks.py [hunk 6]
```diff
@@ -1042,15 +1035,15 @@ def run(cls, filename, dry_run, **kwargs):
     AssessmentAssessmentFeedbackOptionsTask,
     AssessmentAssessmentFeedbackOptionTask,
     AssessmentAssessmentPartTask,
-    AssessmentCriterionTask,
-    AssessmentCriterionOptionTask,
+    # AssessmentCriterionTask,
+    # AssessmentCriterionOptionTask,
```
Comment (0x29a) [inline exporter/tasks.py:1040]: Could you please add a comment about why you've commented this out?

File: requirements.txt [hunk 7]
```diff
@@ -1,20 +1,46 @@
...
+# pymongo==3.10.1
+# python-dateutil==2.4.0
+# pyyaml==5.4.1
+# psutil==5.8.0
+
+# # Tests
+# coverage==5.5
+# mock==4.0.3
+# pytest==6.2.3
+# pytest-cov==2.11.1
```
Comment (0x29a) [inline requirements.txt]: Could you please add a brief comment about why you've commented this out?

File: exporter/tasks.py (hunk 1 above)
Comment (kaustavb12) [reply exporter/tasks.py:296]: With FatalTaskError, the script quits entirely and no output is produced. 
Whereas with Exception, a failure file is generated for the failed task with the extension '.failed'. All other tasks execute unaffected.
Refer [this method](https://github.com/open-craft/edx-analytics-exporter/blob/655d886c2c62b755379ede38e05ab7e50eff5b50/exporter/main.py#L145)

File: exporter/tasks.py (hunk 2 above)
Comment (kaustavb12) [reply exporter/tasks.py:307]: Same as above

File: exporter/tasks.py (hunk 3 above)
Comment (kaustavb12) [reply exporter/tasks.py:313]: Same as above

File: exporter/tasks.py (hunk 5 above)
Comment (kaustavb12) [reply exporter/tasks.py:289]: Thanks :)

File: exporter/tasks.py (hunk 6 above)
Comment (kaustavb12) [reply exporter/tasks.py:1040]: Done

File: requirements.txt (hunk 7 above)
Comment (kaustavb12) [reply requirements.txt]: Done

**Review by 0x29a — APPROVED** [review]
:+1:

- [x] I tested this:
    1. Script exports all courses.
    2. Script exports course data.
    3. Courses can be imported back to the LMS.
- [x] I read through the diff.

**Pull request merged**